### DISCUSSION TITLE
Ability to add custom meta data or front matter using GUI

### DISCUSSION THREAD

**Discussion by thejeshgn** [zim-desktop-wiki/zim-desktop-wiki#2000] [Ideas]
By default the Zim adds the following meta data. I think it will be great to have an ability to add more custom meta data. Things like lang, authors, ISBN etc.  Almost like Key Values set per page.

Now one can add it using external editors. But it will great to add it using GUI

```
Content-Type: text/x-zim-wiki
Wiki-Format: zim 0.6
Creation-Date: 2020-06-11T10:11:00+05:30

```

For example

```
Content-Type: text/x-zim-wiki
Wiki-Format: zim 0.6
Creation-Date: 2020-06-11T10:11:00+05:30
Lang: kn
Authors: Thejesh GN
ISBN: 11823816728
```

Anyway we can access it using [this api](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/c049a59a91c9a608595c9387e3bd4c8b160d6f0e/zim/export/template.py#L426)

**Comment by Nick222** [comment]
I add such "meta-data" in my notes for bibliography - in it`s main space:

Author: J.Smith
Title: Good Book
Year: 2010

And then my own special plugin may get this information and add it in special database.

But how You want to work with Your metadata - by which instruments?
Read? Write? Other?

**Comment by thejeshgn** [comment]
I usually add at the top using the external editor. Movie and book details. 

As of now use in scripts.

**Comment by quazardous** [comment]
Adding a way to edit and exploit meta data for each page could be sweet.

The possibilities are huge:

- Flags in the explorer panel
- Custom/filtered tabular views
- normalized/generated content

**Comment by introt** [comment]
Moved this over to Discussions for further discussion. Related: #1705, #1801

**Comment by quazardous** [comment]
metadata editing and exploiting.

Using zim meta-data format and adding a way to use pages as DB record like items.

### Form template
Using form templates associated with a given metadata (Page-Type or other).
ie: I want to edit book cards so I add a meta data 

```
Page-Type: book
```

And somewhere I link book with a form template describing all the metadata field that can be edited for Books (input type, data type, etc)

ie:

- Book-Author: text edit
- Book-Genre: list of SF,Fantasy,etc


Page-Type could be multi valued to allow aggregated forms.

### Custom Views

Create a panel to list Pages and display custom metadata in columns.
Add filter and sorting capabilities.

### Flags
Use metadata to implements flags in the explorer panel.
ie: Add a flag OK/KO associated with a metadata `Page-Status`.
Add a way to describe a flag (image/metadata/value map).


### Integrate with search

Add possibility to search metadata maybe like github search (ie `user:David`).
